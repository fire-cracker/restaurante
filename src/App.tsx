import React, { FC, useRef, RefObject, useEffect } from 'react'
import { BrowserRouter, Switch, Route } from 'react-router-dom'
import { connect } from 'react-redux'
import Container from 'react-bootstrap/Container'
import { ToastContainer } from 'react-toastify'

import Header from './components/Header'
import LandingPage from './views/LandingPage'
import MenusSection from './components/Menus'
import ReservationPage from './views/ReservationPage'
import { IRootState } from './redux/reducers'
import { IUser, IUserState } from './types/usersTypes'
import { setLoggedInState, getUserProfile } from './redux/actions/users'

interface Props {
  userState: IUserState
  setLoggedInState: (user: IUser) => void
  getUserProfile: (id: string) => any
}

const App: FC<Props> = ({ userState, setLoggedInState, getUserProfile }) => {
  const [modalShow, setModalShow] = React.useState(false)
  const menuRef: RefObject<any> = useRef(null)

  useEffect(() => {
    if (localStorage.token) {
      const { token } = localStorage

      let id: string = ''
      try {
        ({ id } = JSON.parse(window.atob(token.split('.')[1])))
      } catch (error) {
        localStorage.clear()
      }

      if (id) {
        const setUserState = async() => {
          const user = await getUserProfile(id)
          if (user) await setLoggedInState(user)
          else {
            localStorage.clear()
          }
        };
        setUserState()
      }
    }
  }, [getUserProfile, setLoggedInState])

  const scrollToRef = (history: any, ref: any) => {
    if (ref.current) ref.current.scrollIntoView({ behavior: 'smooth', block: 'start' })
    else {
      history.push('/home/menu')
      setTimeout(() => {
        ref.current.scrollIntoView({ behavior: 'smooth', block: 'start' })
      }, 200)
    }
  }

  const executeScroll = (history: any) => {
    scrollToRef(history, menuRef)
  }

  const onModalShow = () => {
    setModalShow(true)
  }

  return (
    <BrowserRouter>
      <ToastContainer />
      <Container fluid className='p-0'>
        <Header executeScrollToMenu={executeScroll} onModalShow={onModalShow} />
        <Switch>
          <Route path='/home' render={(props) => <LandingPage {...props} menuRef={menuRef} modalShow={modalShow} onModalHide={() => setModalShow(false)} />} />
          <Route exact path='/home/menu' render={() => <MenusSection menuRef={menuRef} />} />
          <Route exact path='/reservation' render={() => <ReservationPage />} />
        </Switch>
      </Container>
    </BrowserRouter>
  )
}

const mapStateToProps = (state: IRootState) => ({ userState: state.userState })
const mapDispatchToProps = { setLoggedInState, getUserProfile }

export default connect(mapStateToProps, mapDispatchToProps)(App)
