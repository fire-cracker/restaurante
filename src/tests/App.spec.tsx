import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import App from '../App'

const createMockStore = configureMockStore([thunk])
const store = createMockStore({
  userState: {
    user: null,
    logingIn: false,
    isLoggedIn: false
  },
  menuState: {
    menus: null,
    count: 0,
    fetching: false,
    fetched: false
  }
})

window.scrollTo = jest.fn()
describe('App', () => {
  const props = {}

  let wrapper: any
  beforeAll(() => {
    wrapper = render(
      <Provider store={store}>
        <App {...props} />
      </Provider>
    )
  })

  afterAll(() => {
    jest.clearAllMocks()
  })

  test('renders App component', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
