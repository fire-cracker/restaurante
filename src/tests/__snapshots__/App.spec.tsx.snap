// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="p-0 bg-lavender container-fluid"
      >
        <nav
          class="sticky-top navbar navbar-expand-lg navbar-dark bg-black"
          id="nav"
        >
          <a
            class="navbar-brand"
            href="/"
          >
            Restauranté
          </a>
          <button
            aria-controls="basic-navbar-nav"
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="navbar-collapse collapse"
            id="basic-navbar-nav"
          >
            <div
              class="m-auto main-tabs navbar-nav"
            >
              <a
                class="nav-link active"
                data-rb-event-key="home"
                href="/"
              >
                HOME
              </a>
              <a
                class="nav-link"
                data-rb-event-key="menu"
                href="/menu"
              >
                MENU
              </a>
              <a
                class="nav-link"
                data-rb-event-key="reservation"
                href="/reservation"
              >
                BOOK A TABLE
              </a>
            </div>
            <div
              class="mr navbar-nav"
            >
              <a
                class="btn btn-outline-darkkhaki"
                href="#"
                role="button"
              >
                LOGIN
              </a>
            </div>
          </div>
        </nav>
        <div
          class="home-wrapper p-0 container-fluid"
        >
          <div
            class="p-0 jumbotron jumbotron-fluid"
          >
            <div
              class="container-fluid"
            >
              <div
                class="row"
              >
                <div
                  class="embed-responsive embed-responsive-1by1"
                >
                  <div
                    class="banner d-flex flex-column justify-content-center align-items-center embed-responsive-item col"
                  >
                    <div
                      class="text-0 text-darkkhaki"
                    >
                      <p>
                        Restaurante Exquisite Cuisine
                      </p>
                    </div>
                    <div
                      class="text-1 text-white"
                    >
                      <p
                        class="font-weight-normal"
                      >
                        Enjoy our Food Experience
                      </p>
                    </div>
                    <div
                      class="divider text-darkkhaki d-flex flex-row"
                    >
                      <p
                        class="line my-2 mx-2"
                      >
                         
                      </p>
                      <p
                        class="icon"
                      >
                        <i
                          class="fa fa-cutlery"
                        />
                      </p>
                      <p
                        class="line my-2 mx-2"
                      >
                         
                      </p>
                    </div>
                    <div
                      class="text-2 text-white"
                    >
                      <p>
                        OPENING HOURS
                      </p>
                    </div>
                    <div
                      class="text-3 text-white d-flex flex-row"
                    >
                      <p
                        class="days"
                      >
                        Monday - Friday: 
                      </p>
                      <p
                        class="time text-dimgray"
                      >
                         07:30AM - 23:00PM
                      </p>
                    </div>
                    <div
                      class="text-4 text-white d-flex flex-row"
                    >
                      <p
                        class="days"
                      >
                        Weekends: 
                      </p>
                      <p
                        class="time text-dimgray"
                      >
                         07:30AM - 03:00PM
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="menu-jumbotron bg-ghostwhite px-0 jumbotron jumbotron-fluid"
          >
            <div
              class="menu-container m-auto container"
            >
              <div
                class="container-wrapper px-0 row"
              >
                <div
                  class="container-body m-0 d-flex flex-column justify-content-center align-items-center container"
                >
                  <div
                    class="menu-header mb-4 flex-column justify-content-center align-items-center row"
                  >
                    <div
                      class="col"
                    >
                      <h4
                        class="text-darkkhaki"
                      >
                        FOOD AT FIRST SIGHT
                      </h4>
                    </div>
                    <div
                      class="col"
                    >
                      <h2>
                        Our Restaurant Menu
                      </h2>
                    </div>
                    <div
                      class="divider mt-1 mb-3 text-darkkhaki d-flex flex-row justify-content-center align-items-center col"
                    >
                      <p
                        class="line my-2 mx-1"
                      >
                         
                      </p>
                      <p
                        class="icon"
                      >
                        <img
                          alt="entwined"
                          class="fa fa-cutlery"
                          src="/images/entwined.svg"
                        />
                      </p>
                      <p
                        class="line my-2 mx-1"
                      >
                         
                      </p>
                    </div>
                  </div>
                  <div
                    class="navs-container mb-4 w-100 row"
                  >
                    <div
                      class="w-100 justify-content-between nav"
                      role="tablist"
                    >
                      <div
                        class="nav-col p-0 col-lg-3 col-md-5"
                      >
                        <div
                          class="mr-4 mt-1 mb-0 p-0 nav-item"
                        >
                          <a
                            aria-selected="true"
                            class="py-4 px-3 bg-white nav-link active"
                            data-rb-event-key="breakfast"
                            href="#"
                            role="tab"
                          >
                            <img
                              alt="breafast icon"
                              src="/images/breakfast.svg"
                            />
                            <p>
                              Breakfast
                            </p>
                          </a>
                        </div>
                      </div>
                      <div
                        class="nav-col p-0 col-lg-3 col-md-5"
                      >
                        <div
                          class="mr-4 mt-1 mb-0 p-0 nav-item"
                        >
                          <a
                            aria-selected="false"
                            class="py-4 px-3 bg-white nav-link"
                            data-rb-event-key="lunch"
                            href="#"
                            role="tab"
                            tabindex="-1"
                          >
                            <img
                              alt="breafast icon"
                              src="/images/lunch.svg"
                            />
                            <p>
                              Lunch
                            </p>
                          </a>
                        </div>
                      </div>
                      <div
                        class="nav-col p-0 col-lg-3 col-md-5"
                      >
                        <div
                          class="mr-4 mt-1 mb-0 p-0 nav-item"
                        >
                          <a
                            aria-selected="false"
                            class="py-4 px-3 bg-white nav-link"
                            data-rb-event-key="dinner"
                            href="#"
                            role="tab"
                            tabindex="-1"
                          >
                            <img
                              alt="breafast icon"
                              src="/images/dinner.svg"
                            />
                            <p>
                              Dinner
                            </p>
                          </a>
                        </div>
                      </div>
                      <div
                        class="nav-col mr-0 p-0 col-lg-3 col-md-5"
                      >
                        <div
                          class="mr-0 mt-1 mb-0 p-0 nav-item"
                        >
                          <a
                            aria-selected="false"
                            class="py-4 px-3 bg-white nav-link"
                            data-rb-event-key="drink"
                            href="#"
                            role="tab"
                            tabindex="-1"
                          >
                            <img
                              alt="breafast icon"
                              src="/images/drink.svg"
                            />
                            <p>
                              Drinks
                            </p>
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="menu-body mx-0 pt-5 row"
                  >
                    <div
                      class="tab-content"
                    >
                      <div
                        aria-hidden="false"
                        class="fade tab-pane active show"
                        role="tabpanel"
                      >
                        <div
                          class="row"
                        />
                      </div>
                      <div
                        aria-hidden="true"
                        class="fade tab-pane"
                        role="tabpanel"
                      >
                        <div
                          class="row"
                        />
                      </div>
                      <div
                        aria-hidden="true"
                        class="fade tab-pane"
                        role="tabpanel"
                      >
                        <div
                          class="row"
                        />
                      </div>
                      <div
                        aria-hidden="true"
                        class="fade tab-pane"
                        role="tabpanel"
                      >
                        <div
                          class="row"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="p-0 mb-0 footer-wrapper d-flex justify-content-center align-items-center jumbotron jumbotron-fluid"
        >
          <div
            class="container-fluid"
          >
            <div
              class="footer-banner row"
            >
              <div
                class="text-white px-3 mx-auto row"
              >
                <div
                  class="p-0 d-flex flex-column justify-content-center align-items-center col"
                >
                  <div
                    class="footer-text-0 text-darkkhaki"
                  >
                    <h3>
                      Restauranté
                    </h3>
                  </div>
                  <div
                    class="footer-text-1 text-center text-dimgray"
                  >
                    <p
                      class="font-weight-normal mb-0"
                    >
                      Maryland, South Estate HQ 24 Fifth st., Los Angeles, Nigeria
                    </p>
                    <p
                      class="font-weight-normal mb-0"
                    >
                      Phone: (+01) 123 456 7890
                    </p>
                    <p
                      class="font-weight-normal mb-0"
                    >
                      info@example.com
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="p-0 bg-lavender container-fluid"
    >
      <nav
        class="sticky-top navbar navbar-expand-lg navbar-dark bg-black"
        id="nav"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Restauranté
        </a>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            class="m-auto main-tabs navbar-nav"
          >
            <a
              class="nav-link active"
              data-rb-event-key="home"
              href="/"
            >
              HOME
            </a>
            <a
              class="nav-link"
              data-rb-event-key="menu"
              href="/menu"
            >
              MENU
            </a>
            <a
              class="nav-link"
              data-rb-event-key="reservation"
              href="/reservation"
            >
              BOOK A TABLE
            </a>
          </div>
          <div
            class="mr navbar-nav"
          >
            <a
              class="btn btn-outline-darkkhaki"
              href="#"
              role="button"
            >
              LOGIN
            </a>
          </div>
        </div>
      </nav>
      <div
        class="home-wrapper p-0 container-fluid"
      >
        <div
          class="p-0 jumbotron jumbotron-fluid"
        >
          <div
            class="container-fluid"
          >
            <div
              class="row"
            >
              <div
                class="embed-responsive embed-responsive-1by1"
              >
                <div
                  class="banner d-flex flex-column justify-content-center align-items-center embed-responsive-item col"
                >
                  <div
                    class="text-0 text-darkkhaki"
                  >
                    <p>
                      Restaurante Exquisite Cuisine
                    </p>
                  </div>
                  <div
                    class="text-1 text-white"
                  >
                    <p
                      class="font-weight-normal"
                    >
                      Enjoy our Food Experience
                    </p>
                  </div>
                  <div
                    class="divider text-darkkhaki d-flex flex-row"
                  >
                    <p
                      class="line my-2 mx-2"
                    >
                       
                    </p>
                    <p
                      class="icon"
                    >
                      <i
                        class="fa fa-cutlery"
                      />
                    </p>
                    <p
                      class="line my-2 mx-2"
                    >
                       
                    </p>
                  </div>
                  <div
                    class="text-2 text-white"
                  >
                    <p>
                      OPENING HOURS
                    </p>
                  </div>
                  <div
                    class="text-3 text-white d-flex flex-row"
                  >
                    <p
                      class="days"
                    >
                      Monday - Friday: 
                    </p>
                    <p
                      class="time text-dimgray"
                    >
                       07:30AM - 23:00PM
                    </p>
                  </div>
                  <div
                    class="text-4 text-white d-flex flex-row"
                  >
                    <p
                      class="days"
                    >
                      Weekends: 
                    </p>
                    <p
                      class="time text-dimgray"
                    >
                       07:30AM - 03:00PM
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="menu-jumbotron bg-ghostwhite px-0 jumbotron jumbotron-fluid"
        >
          <div
            class="menu-container m-auto container"
          >
            <div
              class="container-wrapper px-0 row"
            >
              <div
                class="container-body m-0 d-flex flex-column justify-content-center align-items-center container"
              >
                <div
                  class="menu-header mb-4 flex-column justify-content-center align-items-center row"
                >
                  <div
                    class="col"
                  >
                    <h4
                      class="text-darkkhaki"
                    >
                      FOOD AT FIRST SIGHT
                    </h4>
                  </div>
                  <div
                    class="col"
                  >
                    <h2>
                      Our Restaurant Menu
                    </h2>
                  </div>
                  <div
                    class="divider mt-1 mb-3 text-darkkhaki d-flex flex-row justify-content-center align-items-center col"
                  >
                    <p
                      class="line my-2 mx-1"
                    >
                       
                    </p>
                    <p
                      class="icon"
                    >
                      <img
                        alt="entwined"
                        class="fa fa-cutlery"
                        src="/images/entwined.svg"
                      />
                    </p>
                    <p
                      class="line my-2 mx-1"
                    >
                       
                    </p>
                  </div>
                </div>
                <div
                  class="navs-container mb-4 w-100 row"
                >
                  <div
                    class="w-100 justify-content-between nav"
                    role="tablist"
                  >
                    <div
                      class="nav-col p-0 col-lg-3 col-md-5"
                    >
                      <div
                        class="mr-4 mt-1 mb-0 p-0 nav-item"
                      >
                        <a
                          aria-selected="true"
                          class="py-4 px-3 bg-white nav-link active"
                          data-rb-event-key="breakfast"
                          href="#"
                          role="tab"
                        >
                          <img
                            alt="breafast icon"
                            src="/images/breakfast.svg"
                          />
                          <p>
                            Breakfast
                          </p>
                        </a>
                      </div>
                    </div>
                    <div
                      class="nav-col p-0 col-lg-3 col-md-5"
                    >
                      <div
                        class="mr-4 mt-1 mb-0 p-0 nav-item"
                      >
                        <a
                          aria-selected="false"
                          class="py-4 px-3 bg-white nav-link"
                          data-rb-event-key="lunch"
                          href="#"
                          role="tab"
                          tabindex="-1"
                        >
                          <img
                            alt="breafast icon"
                            src="/images/lunch.svg"
                          />
                          <p>
                            Lunch
                          </p>
                        </a>
                      </div>
                    </div>
                    <div
                      class="nav-col p-0 col-lg-3 col-md-5"
                    >
                      <div
                        class="mr-4 mt-1 mb-0 p-0 nav-item"
                      >
                        <a
                          aria-selected="false"
                          class="py-4 px-3 bg-white nav-link"
                          data-rb-event-key="dinner"
                          href="#"
                          role="tab"
                          tabindex="-1"
                        >
                          <img
                            alt="breafast icon"
                            src="/images/dinner.svg"
                          />
                          <p>
                            Dinner
                          </p>
                        </a>
                      </div>
                    </div>
                    <div
                      class="nav-col mr-0 p-0 col-lg-3 col-md-5"
                    >
                      <div
                        class="mr-0 mt-1 mb-0 p-0 nav-item"
                      >
                        <a
                          aria-selected="false"
                          class="py-4 px-3 bg-white nav-link"
                          data-rb-event-key="drink"
                          href="#"
                          role="tab"
                          tabindex="-1"
                        >
                          <img
                            alt="breafast icon"
                            src="/images/drink.svg"
                          />
                          <p>
                            Drinks
                          </p>
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="menu-body mx-0 pt-5 row"
                >
                  <div
                    class="tab-content"
                  >
                    <div
                      aria-hidden="false"
                      class="fade tab-pane active show"
                      role="tabpanel"
                    >
                      <div
                        class="row"
                      />
                    </div>
                    <div
                      aria-hidden="true"
                      class="fade tab-pane"
                      role="tabpanel"
                    >
                      <div
                        class="row"
                      />
                    </div>
                    <div
                      aria-hidden="true"
                      class="fade tab-pane"
                      role="tabpanel"
                    >
                      <div
                        class="row"
                      />
                    </div>
                    <div
                      aria-hidden="true"
                      class="fade tab-pane"
                      role="tabpanel"
                    >
                      <div
                        class="row"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="p-0 mb-0 footer-wrapper d-flex justify-content-center align-items-center jumbotron jumbotron-fluid"
      >
        <div
          class="container-fluid"
        >
          <div
            class="footer-banner row"
          >
            <div
              class="text-white px-3 mx-auto row"
            >
              <div
                class="p-0 d-flex flex-column justify-content-center align-items-center col"
              >
                <div
                  class="footer-text-0 text-darkkhaki"
                >
                  <h3>
                    Restauranté
                  </h3>
                </div>
                <div
                  class="footer-text-1 text-center text-dimgray"
                >
                  <p
                    class="font-weight-normal mb-0"
                  >
                    Maryland, South Estate HQ 24 Fifth st., Los Angeles, Nigeria
                  </p>
                  <p
                    class="font-weight-normal mb-0"
                  >
                    Phone: (+01) 123 456 7890
                  </p>
                  <p
                    class="font-weight-normal mb-0"
                  >
                    info@example.com
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
